### DISCUSSION TITLE
Can Hy have class member annotations?

### DISCUSSION THREAD

**Discussion by fabricio-p** [hylang/hy#2564] [Ideas]
I do the following
```py
@dataclass
class Foo:
  bar: int
```
a lot in python, so I thought this would work
```hy
(defclass [dataclass] Foo
  (annotate bar int))
  ```
but it doesn't, I get the following error:
```
Traceback (most recent call last):
  File "stdin-f43089fa4772bdc914dfef4fe46e95692ddff4a4", line 2
    (annotate bar int))
    ^
hy.errors.HySyntaxError: parse error for pattern macro 'defclass': got unexpected token: hy.models.Expression([
  hy.models.Symbol('annotate'),
  hy.models.Symbol('bar'),
  hy.models.Symbol('int')]), expected: end of macro call
```
I was wondering if there's any plan to support this.

**Comment by Kodiologist** [comment]
`(annotate bar int)` is fine. It's your `defclass` call that's faulty, because it's missing an inheritance list.

**Reply by fabricio-p** [reply]
Oh thanks, I assumed that was optional. While I'm here, can you tell me how I can fix this?
```
ERROR: Could not install packages due to an OSError: [Errno 13] Permission denied: '/home/<user>/.local/get_version'
Check the permissions.
```
I'm doing `pip install --user hy`.

**Reply by Kodiologist** [reply]
I don't think that's related, and I don't know what to tell you beyond what the error message already says.
